### ISSUE TITLE
Point Cloud Initialization

### ISSUE THREAD

**Issue by hyoseok1223** [jaidevshriram/realmdreamer#2]
Hi thanks for great work!

I have a question about implementation detail of  point cloud initialization,
How can align predictions from marigold with the absolute metric depth? ( more detail please )

And have you any mind of sharing your this aligned model weights?

Thanks again for this awesome research!

**Comment by jaidevshriram** [comment]
Thanks for your comments and the question! 

We want to convert an image to point cloud and use Marigold to do this, obtaining $\hat{d}_{\text{marigold}}$. However, marigold predicts depth in \[0, 1\]. Instead, we prefer to make it metric scale. 

Hence, we also predict metric depth using DepthAnything, to obtain  $\hat{d}_{\text{abs}}$.

We find that the detail present in $\hat{d}_{\text{marigold}}$ is much better than

 $\hat{d}_{\text{marigold}}$, so we align the two predictions, by scaling the Marigold depth to match the scale of the metric depth. Here is some code that might help:


```
def get_scale_translation(
        x: np.ndarray, # (N,)
        y: np.ndarray, # (N,)
    ):

    """
    Given two N dimensions tensors, compute a solution to a * x + b = y. Return a, b
    """

    x = x[:, None] # (N, 1)
    x = np.concatenate([x, np.ones(shape=x.shape)], -1) # (N, 2)
    psuedo = np.linalg.inv(x.T @ x) @ x.T

    scale, translation = (psuedo @ y[...,None]).squeeze()

    return scale, translation

def align_depths(
        source_depth: torch.Tensor, # B 1 H W
        target_depth: torch.Tensor, # B 1 H W
        mask: torch.Tensor=None # B 1 H W
    ):

    """
    Given two depth maps, align one with the other. If a mask is provided, will use only the values within the mask
    """

    batch_size, _, _, _ = source_depth.shape

    output_depth = source_depth.clone()

    for i in range(batch_size):

        source_depth_i = source_depth[i].squeeze() # H W
        target_depth_i = target_depth[i].squeeze()

        if mask is not None:
            mask_i = mask[i].squeeze()
            source_depth_i = source_depth_i[mask_i]
            target_depth_i = target_depth_i[mask_i]

        scale, translation = get_scale_translation(source_depth_i.flatten().cpu().detach().numpy(), target_depth_i.flatten().cpu().detach().numpy())

        output_depth[i] = translation + scale * source_depth

    return output_depth
```

**Comment by hyoseok1223** [comment]
Thank you for fast reply!

I prefectly understand that "fit to metric depth" for point cloud initialization!

Thank you!!
